zbus/src/connection/mod.rs

```diff
@@ -1370,6 +1402,112 @@ mod tests {
         let name_has_owner = dbus.name_has_owner(name.try_into().unwrap()).await.unwrap();
         assert!(!name_has_owner);
     }
+
+    #[tokio::test(start_paused = true)]
```

Comment:
* We can't assume the optional tokio feature/dep. This doesn't need to be tokio-specific anyway. We've `future*`, `event-listener`, `async-broadcast` deps as well as async runtime abstractions in `abstractions` module that provide everything you need. The exception would be the `Barrier` but I think it's not necessarily needed.
* Why start_paused?

Reply:
> > We can't assume the optional tokio feature/dep.
> 
> Can't we? It's in dev-dependencies so it's guaranteed to be there for testing regardless of other feature selection.

Let me rephase: we **can** (as in ability) but we shouldn't so that the feature is tested for both async runtime backends.

Reply:
If I've understood correctly, then the fact that tokio is present in dev-dependencies is independent from the feature selection, i.e. it is possible to run this test both with and without tokio as the internal zbus runtime - if the `tokio` feature is not enabled then zbus will still spawn its separate async-io executor for the Connection.

The only sort-of-requirement is that the tokio time-auto-advance feature is useless but harmless if the tight tokio integration is off, as time will immediately advance due to the tokio runtime being idle as the connection is on another runtime - it doesn't actually prove that shutdown doesn't return. But if it's at worst harmless and at best a useful extra check, it seems worth it to me, as the test suite should be run with at least the full feature set, if not again with more combinations. From a quick glance at the CI, it looks like both with and without tokio are indeed tested.

It seems from the other thread that you're on board with this tokio usage now?

Reply:
> If I've understood correctly, then the fact that tokio is present in dev-dependencies is independent from the feature selection, i.e. it is possible to run this test both with and without tokio as the internal zbus runtime - if the `tokio` feature is not enabled then zbus will still spawn its separate async-io executor for the Connection.

:heavy_check_mark: 

> The only sort-of-requirement is that the tokio time-auto-advance feature is useless but harmless if the tight tokio integration is off, as time will immediately advance due to the tokio runtime being idle as the connection is on another runtime - it doesn't actually prove that shutdown doesn't return. But if it's at worst harmless and at best a useful extra check, it seems worth it to me, as the test suite should be run with at least the full feature set, if not again with more combinations. From a quick glance at the CI, it looks like both with and without tokio are indeed tested.

Ah ok, sure thing.
 
> It seems from the other thread that you're on board with this tokio usage now?

:+1: